control-myplaces/src/main/java/fi/nls/oskari/control/myplaces/handler/MyPlacesLayersHandler.java

```diff
@@ -57,131 +60,97 @@ public void preProcess(ActionParameters params) throws ActionException {
 
     @Override
     public void handleGet(ActionParameters params) throws ActionException {
-        final String uuid = params.getUser().getUuid();
-
-        final List<MyPlaceCategory> categories;
+        List<MyPlaceCategory> categories;
         try {
+            String uuid = params.getUser().getUuid();
             categories = layerService.getByUserId(uuid);
+            if (categories.isEmpty()) {
+                // If user has no categories insert a new default category
+                MyPlaceCategory category = insertDefaultCategory(uuid);
+                categories = Collections.singletonList(category);
+            }
         } catch (ServiceException e) {
             LOG.warn(e);
             throw new ActionException("Failed to get myplaces layers");
         }
 
-        if (categories.size() == 0) {
-            // If user has no categories insert a new default category
-            categories.add(insertDefaultCategory(uuid));
-        }
-        OskariLayer baselayer = service.getBaseLayer();
-        ByteArrayOutputStream baos;
-        try {
-            baos = MyPlaceCategoryHelper.toGeoJSONFeatureCollection(categories, baselayer);
-        } catch (IOException e) {
-            LOG.warn(e);
-            throw new ActionException("Failed to create response");
-        }
-
-        ResponseHelper.writeResponse(params, 200,
-                ResponseHelper.CONTENT_TYPE_JSON_UTF8, baos);
+        JSONObject response = toLayersJSON(categories);
+        ResponseHelper.writeResponse(params, 200, response);
     }
 
     @Override
     public void handlePost(ActionParameters params) throws ActionException {
-        final String uuid = params.getUser().getUuid();
-        final List<MyPlaceCategory> categories = readCategories(params, false);
-        for (MyPlaceCategory category : categories) {
-            category.setUuid(uuid);
-        }
-
+        params.requireLoggedInUser();
```

Comment:
This is unnecessary, since `preProcess(ActionParameters params)` already checks this.